File: algorithms/1_easy/M/02016_maximum_difference_between_increasing_elements/solution.c
int maximumDifference(const int *restrict nums, const int len) {
    int max_diff = -1;
    int min = INT_MAX;

    for (int i = 0; i < len; i += 1) {
        if (min >= nums[i]) {
            min = nums[i];
        } else {
            max_diff = MAX(max_diff, nums[i] - min);
        }
    }

    return max_diff;
}

File: algorithms/1_easy/M/02016_maximum_difference_between_increasing_elements/solution.rs
impl Solution {
    pub fn maximum_difference(nums: Vec<i32>) -> i32 {
        let mut max_diff: i32 = -1;
        let mut min: i32 = i32::MAX;

        for num in nums {
            if min >= num {
                min = num;
            } else {
                max_diff = max_diff.max(num - min);
            }
        }

        max_diff
    }
}
